x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/elasticsearch_monitor_states_adapter.ts

```diff
@@ -57,6 +60,26 @@ export class ElasticsearchMonitorStatesAdapter implements UMMonitorStatesAdapter
     };
   }
 
+  public async getSnapshotCount(
+    request: any,
+    dateRangeStart: string,
+    dateRangeEnd: string,
+    filters?: string,
+    statusFilter?: string
+  ): Promise<Snapshot> {
+    const context: QueryContext = {
+      database: this.database,
+      request,
+      dateRangeStart,
+      dateRangeEnd,
+      pagination: CONTEXT_DEFAULTS.CURSOR_PAGINATION,
+      filterClause: filters && filters !== '' ? JSON.parse(filters) : null,
+      size: 30000,
```

Comment:
what is the reason behind size: 3000?

Reply:
It actually says `30,000`.  I think the goal here is to provide an upper bound on the amount of time spent in this query. Is that correct @justinkambic ? Either way a comment would be nice. 

I'd be curious to know how we arrived at 30k either way.

Reply:
This is actually the oldest code in the diff, I wrote it quite some time ago and I can't say for sure but I'm willing to bet I chose 30000 thinking I'd come back to it once I had everything working and extract it to a descriptive comment.

The goal was to provide an upper bound, but I'm not sure if 30k is the right number for that.

Reply:
We could actually make it unlimited by patching the iterator, or set it to something truly silly like 1,000,0000

Reply:
I'm fine with hardcoding this in a constant at some absurd number for this PR and creating a follow-up issue to make the iterator go _ad infinitum_ later. I'd prefer not to touch any more modules in this change.